editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/bean/AttributeTextParameterPropertyModifyListenerForAdvanced.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.properties.bean;
+
+import org.fusesource.ide.camel.editor.properties.creators.modifylisteners.text.AbstractTextParameterPropertyModifyListener;
+import org.fusesource.ide.camel.model.service.core.catalog.Parameter;
+import org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement;
+import org.fusesource.ide.foundation.core.util.Strings;
+import org.w3c.dom.Element;
+
+/**
+ * @author brianf
+ *
+ */
+public class AttributeTextParameterPropertyModifyListenerForAdvanced extends AbstractTextParameterPropertyModifyListener {
+
+	private Parameter parameter;
+
+	public AttributeTextParameterPropertyModifyListenerForAdvanced(AbstractCamelModelElement camelModelElement, Parameter parameter) {
+		super(camelModelElement, parameter.getName());
+		this.parameter = parameter;
+	}
+
+	@Override
+	protected void updateModel(String newValue) {
+		setAttributeValue(this.parameter.getName(), newValue);
+	}
+	
+	private void setAttributeValue(String attrName, String attrValue) {
```

Comment:
yeah, there some false positive how this kind of issue, I reported it and it should be fixed in next version of Sonar. See here for more details: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/sonarqube/ZbizF_J2fD4/oE5lyB-rBgAJ